File: go-sdk/pkg/models/operations/deletestoragepolicyassignmentsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteStoragePolicyAssignmentsIDPathParams struct {
	StoragePolicyAssignmentID string `pathParam:"style=simple,explode=false,name=storage_policy_assignment_id"`
}

type DeleteStoragePolicyAssignmentsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteStoragePolicyAssignmentsIDPathParams
}

type DeleteStoragePolicyAssignmentsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/deletestoragepolicyassignmentsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteStoragePolicyAssignmentsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=storage_policy_assignment_id" })
  storagePolicyAssignmentId: string;
}


export class DeleteStoragePolicyAssignmentsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteStoragePolicyAssignmentsIdPathParams;
}


export class DeleteStoragePolicyAssignmentsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
